service/history/api/updateworkflow/api.go

```diff
@@ -231,15 +234,15 @@ func (u *Updater) OnSuccess(
 		// Speculative WFT was created and needs to be added directly to matching w/o transfer task.
 		// TODO (alex): This code is copied from transferQueueActiveTaskExecutor.processWorkflowTask.
 		//   Helper function needs to be extracted to avoid code duplication.
-		err := u.addWorkflowTaskToMatching(ctx, u.wfKey, u.taskQueue, u.scheduledEventID, u.scheduleToStartTimeout, u.directive)
+		err := u.addWorkflowTaskToMatching(ctx)
```

Comment:
I can't recall why we pass all these `Updater` fields to `Updater` method. @stephanos, is there a specific reason? Because this change makes perfect sense to me.

Reply:
No, I don't see one. I must have transferred them from vars to fields and missed that obvious refactoring.

Reply:
Here: https://github.com/temporalio/temporal/commit/e819c3a57b9e8da67d96c878f46448b8283b38ed